lib/matplotlib/tests/test_axes.py

```diff
@@ -7846,6 +7846,19 @@ def test_centered_bar_label_nonlinear():
     ax.set_axis_off()
 
 
+def test_centered_bar_label_label_beyond_limits(tmp_path):
+    _, ax = plt.subplots()
+
+    last = 0
+    for label, value in zip(['a', 'b', 'c'], [10, 20, 50]):
+        bar_container = ax.barh('col', value, label=label, left=last)
+        ax.bar_label(bar_container, label_type='center')
+        last += value
+    ax.set_xlim(None, 20)
+
+    plt.savefig(tmp_path / 'test')
```

Comment:
I don't see any reason to save this to a file? Do you just need `fig.draw_without_rendering()`?